api/v1/registry.proto

```diff
@@ -140,6 +140,42 @@ message LayerOption {
     string helm_values = 5;
     // Optional set of parameters to be applied to the layer.
     repeated Parameter parameters = 6;
+    // Optional set of cluster requirements for this layer option.
+    // Interfaces to the renderer can implement validations against these requirements.
+    // TODO does this make more sense on the layer itself?
+    repeated EnvironmentRequirements environment_requirements = 7;
+}
+
+message EnvironmentRequirements {
+    // Optional set of resources that must be present in the cluster for this layer to function correctly.
+    // Note that these resources do not factor into manifest rendering, but can be used by interfaces to the
+    // renderer to validate or create expected resources on the cluster before install.
+    repeated ResourceDependency resource_dependencies = 1;
+}
+
+// Represents a resource that must be present on a cluster for install to succeed.
+message ResourceDependency {
+    // Describes a Kubernetes Secret
+    message Secret {
+        // Corresponds to the name on the object metadata of the secret.
+        // Note that referenced secrets must be in the Service Mesh Hub operator's install namespace.
```

Comment:
Is it the operator's install namespace, or the install namespace in the application state CRD?